1.10/deploying-services/private-docker-registry.md

```diff
@@ -71,3 +74,199 @@ To supply credentials to pull from a private Docker registry, create an archive
     ```
 
 1. The Docker image will now pull using the security credentials you specified.
+
+<a name="secret-store-instructions"></a>
+# Referencing private Docker registry credentials in the secrets store (Enterprise DC/OS only)
+
+Follow these steps to add your Docker registry credentials to the Enterprise DC/OS secrets store, and then reference that secret in your service definition.
+
+**Note:** This functionality is only available with the [Universal Containerizer Runtime](/docs/1.10/deploying-services/containerizers/ucr/). If you need to use the Docker Containerizer, follow the [URI instructions](#uri-instructions) above.
+
+## Step 1: Create a credentials file
+
+1.  Log in to your private registry manually. This will create a `~/.docker` directory and a `~/.docker/config.json` file.
+
+    ```bash
+    docker login some.docker.host.com
+    Username: foo
+    Password:
+    Email: foo@bar.com
+    ```
+
+1.  Check that you have the `~/.docker/config.json` file.
+
+    ```bash
+    ls ~/.docker
+    config.json
+    ```
+
+    Your `config.json` file should look like this, where value of `auth` is a base64-encoded `username:password` string.
+
+    ```json
+    {
+      "auths": {
+          "https://index.docker.io/v1/": {
+              "auth": "XXXXX",
+              "email": "<your-email>"
+          }
+      }
+    }
+    ```
+
+    **Note:** If you are using Mac OSX, you will need to manually encode your `username:password` string and modify your `config.json` to match the snippet above.
+
+1. Add the `config.json` file to the DC/OS secret store. [Learn more about creating secrets](https://docs.mesosphere.com/1.9/security/secrets/create-secrets/).
+
+   **Note:** As of DC/OS version 10.0, you can only add a file to the secret store via the DC/OS CLI.
+
+   ```bash
+   dcos security secrets create --value-file=config.json <path/to/secret>
+   ```
+
+   If you plan to follow the example below, enter the following command to add the secret:
+
+   ```bash
+   dcos security secrets create --value-file=config.json mesos-docker/pullConfig
+   ```
+
+## Step 2: Add the secret to your service or pod Definition
+
+### For a service
+
+Add the following two parameters to your service definition.
+
+1.  A location for the secret in the `secrets` parameter:
```

Comment:
This procedure doesn't flow very well. Steps 1 and 2 are showing excerpts of the app definition, then step 3 is showing an entire app definition?

I would rework to say: `Add the `secrets` and `containers` parameters to your pod definition.` and then provide the full example. Also, step 5 is unnecessary. This can be included with the current step 4.